kepler-companion/KeplerCompanion.ts

```diff
@@ -0,0 +1,114 @@
+import getMAC from 'getmac';
+import * as crypto from 'crypto';
+import { Http, Kuzzle } from 'kuzzle-sdk';
+
+export type KeplerCompanionConfiguration = {
+  host?: string,
+  port?: number,
+  ssl?: boolean,
+  trackingPath?: string,
+  enabled?: boolean,
+};
+
+export type TrackingOpts = {
+  action: string,
+  product: string,
+  version: string,
+  tags?: { [key: string]: any },
+}
+
+export default class KeplerCompanion {
+  private sdk: Kuzzle;
+  public config: KeplerCompanionConfiguration = {
+    host: 'analytics.app.kuzzle.io',
+    port: 443,
+    ssl: true,
+    trackingPath: '/_/analytics/track',
+    enabled: true,
+  };
+
+  constructor(config = {}) {
+    this.config = { ...this.config, ...config };
+    this.sdk = new Kuzzle(
+      new Http(
+        this.config.host, 
+        { port: this.config.port, ssl: this.config.ssl }
+      )
+    );
+  }
+
+  private getUserId(mode: 'node' | 'browser', product: string): string {
+    let id: string;
+    switch (mode) {
+      case 'node':
+        id = crypto.createHash('sha256').update(getMAC()).digest('hex');
+      break;
+      case 'browser':
+        id = window.localStorage.getItem(`${product}-kepler-id`);
+        if (! id) {
+          id = generateUniqueIdForBrowser();
+          window.localStorage.setItem(`${product}-kepler-id`, id);
+        }
+      break;
+    }
+
+    return id;
+  }
+
+  public turnOff() {
+    this.config.enabled = false;
+  }
+
+  public track(opts: TrackingOpts, timeout = 1000) {
+    if (!this.config.enabled) {
+      return;
+    }
+
+    if (process.env.CI) {
+      opts.tags = opts.tags || {};
+      opts.tags.ci = true;
+    }
+
+    const innerTrack = this._track.bind(this);
+    return Promise.race([
+      new Promise(() => {
+        try {
+          return innerTrack(opts);
+        } catch (_) { /* Analytics should not interfer with user process */ }
+      }),
```

Comment:
Should be replaced by (also, you are not awaiting the promise)
```suggestion
      innerTrack(opts).catch(() => {});
```